test/abstractarray.jl

```diff
@@ -912,3 +912,8 @@ end
     @test hcat(1:2, fill(1, (2,1))) == hcat([1:2;], fill(1, (2,1))) == reshape([1,2,1,1],2,2)
     @test [(1:3) (4:6); fill(1, (3,2))] == reshape([1,2,3,1,1,1,4,5,6,1,1,1], 6,2)
 end
+
+@testset "issue #27138" begin
+    @test convert(AbstractVector{Float64},1:10) === 1.0:1.0:10.0
```

Comment:
@nalimilan this test would break, if `convert` would not fall back to `AbstractVector{T}(r::AbstractRange)`.

Reply:
OK, got it, due to the fallback method `convert(::Type{AbstractArray{T,N}}, a::AbstractArray{<:Any,N}) where {T,N} = AbstractArray{T,N}(a)`, `convert` ends up calling the constructor you added.